### DISCUSSION TITLE
Is it possible to use nix module system within nixvim standalone?

### DISCUSSION THREAD

**Discussion by HeitorAugustoLN** [nix-community/nixvim#2034] [Q&A] [answered]
I want to create custom options within a nixvim standalone flake, so when it is consumed by my nixos configuration for example, i can customize it depending on the setup

**Comment by MattSturgeon** [comment]
Nixvim is built on the module system, so yes you can create your own options and define conditional config (etc).

If you wish to modify a standalone nixvim after the fact (e.g. in you nixos config), you can do this using the [`.extend` function](https://nix-community.github.io/nixvim/modules/standalone.html#extending-an-existing-configuration) made available on the standalone derivation.

**Reply by HeitorAugustoLN** [reply]
That's nice! I would have to use `.extend` like this in my config?

```nix
{inputs, system, ...}: {
  inputs.nixvim-config.packages.${system}.default.extend {
    option = true;
  };

  home.packages = [inputs.nixvim-config.packages.${system}.default];
}
```

**Reply by MattSturgeon** [reply, marked as answer]
No, you could do:

```nix
{inputs, pkgs, ...}: {
  home.packages = [
    (inputs.nixvim-config.packages.${pkgs.stdenv.hostPlatform.system}.default.extend {
      option = true;
    })
  ];
}
```

Or:


```nix
{inputs, pkgs, ...}:
let
  inherit (pkgs.stdenv.hostPlatform) system;
  myNvim = inputs.nixvim-config.packages.${system}.default.extend {
    option = true;
  };
in
{
  home.packages = [ myNvim ];
}
```

**Reply by MattSturgeon** [reply]
Incidentally, you could also have your standalone nixvim within the same flake if you prefer. You'd just access it via `self` instead of `inputs`.

**Reply by HeitorAugustoLN** [reply]
Yeah, that makes more sense than the way I thought. Thank you!
